### DISCUSSION TITLE
Google cloud: Blocking China network egress traffic (discussion HOWTO, experience)

### DISCUSSION THREAD

**Discussion by emp-00** [NightscoutFoundation/xDrip#3062] [Ideas]
I know the xDrip team does not want to implement this functionality in the official package due to code complexity, understood. Can we still discuss this here in the forum? 

Q: Has anybody successfully implemented e.g. this "free" solution --> see HOWTO and scripts at the bottom of the page?
https://stackoverflow.com/questions/58699178/how-to-block-egress-trafic-to-china-and-apac-on-gcp

**Comment by Navid200** [comment]
There are several suggestions.  This is one example:  https://github.com/gcp-services/documentation/blob/main/guides/block-egress.md

Just to explain why we have put this on hold:
It's a very different concept if an individual has a virtual machine and wants to block traffic to China versus if there is a project that is used internationally by many people and the group wants to add this capability.

1- There is a requirement to install locales.    A pop-up may appear requiring the user to confirm.  We already have people making mistakes following the installation instructions as is.  Complicating it further is not a good idea.
So, we will need to find a way to ensure a pop-up will never occur.

2- We cannot block traffic to China for everyone.  Our setup can be used by someone who lives in china.  Such a person should not block traffic to China.  Therefore, we will need to add a switch that the user can enable or disable.

3- No matter how we implement all of this, our code will become more complicated than it is now.  Any addition to the code will need to be maintained as we update node, or Ubuntu or Nightsocut, or add more features.  Every time, we make a change, we will need to test all functionalities.    In this case, we will also need to test the block to China every time we make a change.

4- A cron job is required to run daily to fetch updated ip addresses.

This is not a one-liner solution.  It is quite complicated.
All of that to save $0.02 per month may not be worth it.  I have attempted this multiple times.  Every time I start putting a submission together, as I have seen the complexity required, I could not help questioning myself "all of this for $0.02 per month"?
That's the reason as much as we would like to add this feature, looking at the pros and cons, we cannot convince ourselves that it is what we should be doing.  
At least, we can say that we have put this on hold for now until we address other tasks that have higher priority.
